### DISCUSSION TITLE
CompileException not found

### DISCUSSION THREAD

**Discussion by LubieKakao1212** [cc-tweaked/CC-Tweaked#1752] [Q&A] [answered]
I have a Development Environment set up with architectury, when I try to launch the forge version it works fine until I try joining a world then it crashes with:
```
java.lang.ClassNotFoundException: org.squiddev.cobalt.compiler.CompileException
```

the project files can be found here
https://github.com/LubieKakao1212/OpenCombatUtility/tree/dev1.19.4-arch

**Comment by SquidDev** [comment]
Does it work if you remove the dependency on Cobalt?

**Reply by LubieKakao1212** [reply]
No it doesn't.
I added that dependency as an attempt at fixing it

**Reply by SquidDev** [reply, marked as answer]
Right, after some further poking this appears to be a bug with Architectury Loom (have opened https://github.com/architectury/architectury-loom/issues/201)

On normal ForgeGradle, CC:T's additional dependencies are pulled in via jar-in-jar, to ensure they're correctly loaded despite not being mods. However, Architectury strips out the jarjar metadata (https://github.com/architectury/architectury-loom/pull/159), so the jars are entirely ignored[^1]

As a workaround, you can add this to your `forge/build.gradle` file:

```groovy
dependencies {
    forgeRuntimeLibrary "org.squiddev:Cobalt:0.7.3"
    forgeRuntimeLibrary "com.jcraft:jzlib:1.1.3"
    forgeRuntimeLibrary "io.netty:netty-codec-http:4.1.82.Final"
    forgeRuntimeLibrary "io.netty:netty-codec-socks:4.1.82.Final"
    forgeRuntimeLibrary "io.netty:netty-handler-proxy:4.1.82.Final"
}
```

[^1]: They're still on the classpath as they're registered as Maven deps, but not loaded because FML/BSL ignores non-mods by default.

**Reply by LubieKakao1212** [reply]
Thanks for the replay, it works
